### PR TITLE
feat: chore infrastructure for the Repairnator Jenkins plugin

### PR DESCRIPTION
**Pull request by henry-lp**
Making a repairnator plugin for Jenkins - issue (#469)

### CODE REVIEW

**Comment by monperrus** [comment]
nice to see this WIP!

**Comment by henry-lp** [comment]
@monperrus Hi , this is ready for merge :) when travis is done.

Next: change pom.xml in the plugin to download latest snapshots instead after this PR is merged.

File: repairnator-jenkins-plugin/src/main/java/io/jenkins/plugins/main/RepairnatorPostBuild.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    }
+
+    public void printAllEnvVars(AbstractBuild build,BuildListener listener) {
+        try {
+            EnvVars env = build.getEnvironment(listener);
+            for (String key : env.keySet()) {
+                System.out.println(key + " : " + env.get(key));
+            }
+        }catch (Exception e ) {
+            e.printStackTrace(System.out);
```
Comment (monperrus) [inline repairnator-jenkins-plugin/src/main/java/io/jenkins/plugins/main/RepairnatorPostBuild.java]: never e.printStackTrace, prefer `throw new RuntimeException(e)`

File: repairnator-jenkins-plugin/src/main/java/io/jenkins/plugins/main/RepairnatorPostBuild.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+import java.io.PrintStream;
+
+import java.util.logging.*;
+import java.io.IOException;
+
+import java.util.Map;
+import hudson.model.Computer;
+import java.util.Arrays;
+
+public class RepairnatorPostBuild extends Recorder {
```
Comment (monperrus) [inline repairnator-jenkins-plugin/src/main/java/io/jenkins/plugins/main/RepairnatorPostBuild.java]: javadoc

File: repairnator-jenkins-plugin/src/main/java/io/jenkins/plugins/main/RunPipelineAction.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.Enumeration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.logging.Handler;
+import java.util.concurrent.TimeUnit;
+import java.lang.InterruptedException;
+import java.util.logging.Handler;
+
+
+public class RunPipelineAction {
```
Comment (monperrus) [inline repairnator-jenkins-plugin/src/main/java/io/jenkins/plugins/main/RunPipelineAction.java]: javadoc

File: repairnator-jenkins-plugin/src/main/java/io/jenkins/plugins/main/RunPipelineAction.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    public void run() {
+        try {
+            JenkinsLauncher launcher = new JenkinsLauncher();
+            RepairToolsManager.getInstance().manualLoadRepairTools();
+            this.pipeAllLoggerToStderr();
+            launcher.jenkinsMain(this.gitUrl,this.gitToken,this.gitBranch,"NPEFix");
+        } catch(Exception e) {
+            e.printStackTrace(System.out);
```
Comment (monperrus) [inline repairnator-jenkins-plugin/src/main/java/io/jenkins/plugins/main/RunPipelineAction.java]: throw new Runtime

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/repair/AbstractRepairStep.java [hunk 5]
```diff
@@ -174,13 +176,11 @@ protected void createPullRequest(List<File> patchList, int nbPatch) throws IOExc
...
                 String head = ghForkedRepo.getOwnerName() + ":" + branchName;
-
-                String travisURL = Utils.getTravisUrl(this.getInspector().getBuggyBuild().getId(), this.getInspector().getRepoSlug());
-
-                String prText = String.format(TEXT_PR, travisURL, this.getInspector().getRepoSlug(), this.getInspector().getRepoSlug());
-
+                String travisURL = this.getInspector().getBuggyBuild() == null ? "" : Utils.getTravisUrl(this.getInspector().getBuggyBuild().getId(), this.getInspector().getRepoSlug());
+                String baseString = this.getInspector().getBuggyBuild() == null ? "This PR has been created automatically by [repairnator](https://github.com/eclipse/repairnator).\n" : TEXT_PR;
+                String prText = String.format(baseString, travisURL, this.getInspector().getRepoSlug(), this.getInspector().getRepoSlug());
                 GHPullRequest pullRequest = originalRepository.createPullRequest("Patch proposal", head, base, prText);
```
Comment (monperrus) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/repair/AbstractRepairStep.java]: PR Header should be "Patch found by Repairnator with repair tool"+toolName

PR description should contain the name+link of the failing PR/Build, with the failing test names that are fixed.

**Comment by henry-lp** [comment]
@monperrus  jenkins plugin folder is removed . This need to be merged for snapshot per discussion :)

**Comment by monperrus** [comment]
AcK. Commented out the ill test because we need passing CI for merging and deploying a new snapshot.

**Comment by henry-lp** [comment]
test passed , @monperrus merged ? :)

**Comment by monperrus** [comment]
Snapshot will be deployed in 40 minutes.

Thanks a lot for the PR!

**Pull request merged**